lib/jellyfish_web/server_socket.ex

```diff
@@ -115,6 +110,27 @@ defmodule JellyfishWeb.ServerSocket do
     {:stop, :closed, {1003, "operation not allowed"}, state}
   end
 
+  defp handle_request({:room_state_request, %RoomStateRequest{content: {_variant, option}}}) do
+    room_state = get_room_state(option)
+
+    %ServerMessage{content: room_state}
+    |> ServerMessage.encode()
+    |> then(&{:ok, &1})
+  end
+
+  defp handle_request({:subscribe_request, %SubscribeRequest{event_types: event_types}}) do
+    event_types
+    |> Enum.each(fn event ->
+      :ok = Phoenix.PubSub.subscribe(Jellyfish.PubSub, event_type_to_topic(event))
+    end)
+
+    %ServerMessage{content: {:subscription_response, %SubscriptionResponse{}}}
+    |> ServerMessage.encode()
+    |> then(&{:ok, &1})
+  end
+
+  defp handle_request(request), do: request
+
```

Comment:
I would remove the default case all together, or at least rise, we don't expect any other messages.

Reply:
We do raise, in the `handle_in` callback.

In this clause we return `request` instead of `{:ok, mesage}` so we raise in the `with`.
However this isn't explicit really.

Reply:
I agree with @LVala on this -- I'd say either remove the default case, explicitly raise, or return something like `{:error, :unknown_request}` (and maybe log an error); either of these solutions is fine by me